### PR TITLE
resource/resource_alicloud_db_instance: Added retry stragety and Instance status verification;resource/resource_alicloud_db_instance: Fix payment type conversion;resource/resource_alicloud_db_instance_test: Repair automated testing;

### PR DESCRIPTION
**Pull request by zcc01029273**

### CODE REVIEW

File: alicloud/resource_alicloud_db_instance.go [hunk 1]
```diff
@@ -819,6 +832,10 @@ func resourceAlicloudDBInstanceUpdate(d *schema.ResourceData, meta interface{})
 		if err != nil {
 			return WrapErrorf(err, DefaultErrorMsg, d.Id(), action, AlibabaCloudSdkGoERROR)
 		}
+		stateConf := BuildStateConf([]string{}, []string{"Running"}, client.GetRetryTimeout(d.Timeout(schema.TimeoutUpdate)), 3*time.Minute, rdsService.RdsDBInstanceStateRefreshFunc(d.Id(), []string{"Deleting"}))
```
[the remark this answers is not in the document]
Comment (zcc01029273) [reply alicloud/resource_alicloud_db_instance.go:836]: done

File: alicloud/resource_alicloud_db_instance.go [hunk 2]
```diff
@@ -1114,6 +1135,10 @@ func resourceAlicloudDBInstanceUpdate(d *schema.ResourceData, meta interface{})
 			return WrapErrorf(err, DefaultErrorMsg, d.Id(), action, AlibabaCloudSdkGoERROR)
 		}
 		addDebug(action, response, request)
+		stateConf := BuildStateConf([]string{}, []string{"Running"}, d.Timeout(schema.TimeoutUpdate), 0, rdsService.RdsDBInstanceStateRefreshFunc(d.Id(), []string{"Deleting"}))
```
Comment (super-eggs) [inline alicloud/resource_alicloud_db_instance.go:1138]: client.GetRetryTimeout(d.Timeout(schema.TimeoutUpdate)

Comment (zcc01029273) [reply alicloud/resource_alicloud_db_instance.go]: done

File: alicloud/resource_alicloud_db_instance.go [hunk 3]
```diff
@@ -1490,11 +1514,23 @@ func resourceAlicloudDBInstanceRead(d *schema.ResourceData, meta interface{}) er
 		if err != nil {
 			return WrapError(err)
 		}
-		response, err := conn.DoRequest(StringPointer(action), nil, StringPointer("POST"), StringPointer("2014-08-15"), StringPointer("AK"), nil, request, &runtime)
+		var response map[string]interface{}
+		wait := incrementalWait(3*time.Second, 3*time.Second)
+		err = resource.Retry(5*time.Minute, func() *resource.RetryError {
```
Comment (super-eggs) [inline alicloud/resource_alicloud_db_instance.go:1519]: 5*time.Minute => client.GetRetryTimeout(d.Timeout(schema.TimeoutUpdate)

Comment (zcc01029273) [reply alicloud/resource_alicloud_db_instance.go:1519]: 这里是read

File: alicloud/resource_alicloud_db_instance.go [hunk 4]
```diff
@@ -645,21 +645,22 @@ func resourceAlicloudDBInstanceUpdate(d *schema.ResourceData, meta interface{})
 	}
 
 	payType := PayType(d.Get("instance_charge_type").(string))
-	if !d.IsNewResource() && d.HasChange("instance_charge_type") && payType == Prepaid {
-		action := "ModifyDBInstancePayType"
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_db_instance.go:649]: payType == Prepaid  这个判断为啥要去掉，现在是支持预付费转后付费了？

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: alicloud/resource_alicloud_db_instance.go (hunk 4 above)
Comment (zcc01029273) [reply alicloud/resource_alicloud_db_instance.go:649]: api可以支持，前段时间郭志远给我提了个需求就是预付费转后付费的需求aone：46044433

**Pull request merged**